### PR TITLE
Settings file upload

### PR DESCRIPTION
**Pull request by orzechdev**
It adds settings file upload in the customization page.

Please merge first https://github.com/saleor/saleor-checkout/pull/86.

### CODE REVIEW

File: apps/checkout-app/frontend/components/templates/CustomizationDetails/CustomizationDetails.tsx [hunk 1]
```diff
@@ -72,13 +83,31 @@ const CustomizationDetails: React.FC<CustomizationDetailsProps> = ({
...
 
-  const handleSubmit = (flattenedSettings: Record<string, string>) => {
-    onSubmit(
+  const handleFileChange = <T extends CustomizationID>(
+    optionId: T,
+    settingId: CustomizationSettingID<T>,
+    event: React.ChangeEvent<HTMLInputElement>
+  ) => {
+    const inputFiles = event.target.files;
+    if (inputFiles && inputFiles.length > 0) {
```
Comment (bmigirl) [inline apps/checkout-app/frontend/components/templates/CustomizationDetails/CustomizationDetails.tsx:92]:
```suggestion
    if (!!inputFiles?.length) {
```

File: apps/checkout-app/pages/customization.tsx [hunk 2]
```diff
@@ -37,12 +43,59 @@ const Customization = () => {
...
-  const handleSubmit = (data: CustomizationSettingsValues) => {
+  const uploadSettingsFiles = async (
+    data: CustomizationSettingsValues,
+    dataFiles?: CustomizationSettingsFiles
+  ) => {
+    if (!dataFiles) {
+      return data;
+    }
+
+    return await reduce(
```
Comment (bmigirl) [inline apps/checkout-app/pages/customization.tsx:54]: This part be done better:
- raw reduce inside of reduce is hard to get 
- looking at the length of this function, it probably should be a util function, possibly two
- i don't think we should `await` noun-based variables. So we're awaiting fileUpload (that makes sense). What does it mean to await settings or settings urls?
- I'd much rather see the Promise.resolve done first, then put as reduce's base object. It'd be clearer what it actually is. Sometimes it's better to make it a bit longer but keep some of the stuff named as consts

**Review by bmigirl — CHANGES_REQUESTED** [review]

**Comment by jwm0** [comment]
Re-request for review when conflicts are cleaned up

File: apps/checkout-app/pages/customization.tsx (hunk 2 above)
Comment (orzechdev) [reply apps/checkout-app/pages/customization.tsx]: I updated it a bit + also added `Promise.all` to concurrently send files.

**Review by bmigirl — APPROVED** [review]

**Review by jwm0 — APPROVED** [review]

**Pull request merged**